File: Homework6/fileManager/writer/impl/CsvPhoneBookWriter.java
package Homework6.fileManager.writer.impl;


import Homework6.fileManager.writer.PhoneBookWriter;
import Homework6.model.Contact;

import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;

public class CsvPhoneBookWriter implements PhoneBookWriter {

    private PrintWriter writer;


    public CsvPhoneBookWriter(String file) {
        try {
            this.writer = new PrintWriter(new FileWriter(file, true));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public void save(Contact phoneBook) {
        writer.println("\n");
        writer.printf("%s %s: %s", phoneBook.getName(), phoneBook.getSurname(), phoneBook.getPhone());
        writer.flush();
    }


}

File: Homework7/fileManager/writer/impl/CsvPhoneBookWriter.kt
package Homework7.fileManager.writer.impl

import Homework7.fileManager.writer.PhoneBookWriter
import Homework7.model.Contact
import java.io.FileWriter
import java.io.IOException
import java.io.PrintWriter

class CsvPhoneBookWriter(file: String?) : PhoneBookWriter {
    private var writer: PrintWriter? = null

    init {
        try {
            writer = PrintWriter(FileWriter(file, true))
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }

    override fun save(contact: Contact) {
        writer!!.println("\n")
        writer!!.printf("%s %s: %s", contact.name, contact.surname, contact.phone)
        writer!!.flush()
    }
}